BLSQ/iaso
# IA-2739: Add "User" filter on Completeness stats page

Explain what problem this PR is resolving
- Add "User" filter on Completeness stats page
Related JIRA tickets : [IA-2739](https://bluesquare.atlassian.net/browse/IA-2739)

## Self proofreading checklist


## Doc
- Tell us where the doc can be found (docs folder, wiki, in the code...)

## Changes
- Add asyncSelect component on completeness stats filter
- Filter on users in the backend

## How to test
- Create submissions using differents users
- Filter on users and check the returned submissions

## Print screen / video
[Screencast from 2024-03-20 16-37-40.webm](https://github.com/BLSQ/iaso/assets/19631540/ef661012-cd7b-40f8-81ee-dc5f2aefb543)


[IA-2739]: https://bluesquare.atlassian.net/browse/IA-2739?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
